### PR TITLE
Add defination of Strategy pattern

### PR DESCRIPTION
**Pull request by jaskaransingh9990**

### CODE REVIEW

**Comment by BasranS** [comment]
Your work seems good! Just a suggestion to add some examples or scenarios, which could enhance the understandability of the topic well.

**Comment by jaskaransingh9990** [comment]
hey @BasranS  your idea  seems good. i see want can i add thanks

File: README.md [hunk 1]
```diff
@@ -66,11 +66,50 @@ Observer pattern. (2022, November 13). Wikipedia. https://en.wikipedia.org/wiki/
...
+* Context: This class may define an interface that allows the strategy to access its data and holds a reference to a strategy object. The strategy object receives the algorithm from the context class.
+
+* Strategy Interface: An interface or abstract class that defines a collection of methods to carry out a specific behaviour is called a strategy. Using a particular algorithm, each concrete method puts this interface into practice.
+
+
+* Concrete Strategies: These are the ways in which the strategy interface is put into practice. For every concrete strategy, there is a unique algorithm that the context can apply.
+
+### Benifits of using Strategy pattern
+
+* Flexibility and Extensibility:
```
Comment (ash-teach) [inline README.md]: Remove the colon at the end of this bullet

File: README.md [hunk 2]
```diff
@@ -66,11 +66,50 @@ Observer pattern. (2022, November 13). Wikipedia. https://en.wikipedia.org/wiki/
...
+
+
+* Concrete Strategies: These are the ways in which the strategy interface is put into practice. For every concrete strategy, there is a unique algorithm that the context can apply.
+
+### Benifits of using Strategy pattern
+
+* Flexibility and Extensibility: 
+* Promotes Reusability
+* Enhances Testability
+* Easy to  Maintenance
```
Comment (ash-teach) [inline README.md]: "Easy to maintain"

File: README.md [hunk 3]
```diff
@@ -66,11 +66,50 @@ Observer pattern. (2022, November 13). Wikipedia. https://en.wikipedia.org/wiki/
...
+### Real life example
+
+
+Real-World Analogy of Strategy Design Pattern: Payment Options After Shopping in a Mall
+
+Context: You've finished shopping at a mall and need to choose the most convenient method to pay for your purchases.
+
+Strategies:
+ * Credit/Debit Card: 
+ * Cash:
```
Comment (ash-teach) [inline README.md]: I don't think you've done quite enough in this example. It doesn't quite help make sense of Strategy.

File: README.md [hunk 4]
```diff
@@ -66,11 +66,50 @@ Observer pattern. (2022, November 13). Wikipedia. https://en.wikipedia.org/wiki/
...
+Context: You've finished shopping at a mall and need to choose the most convenient method to pay for your purchases.
+
+Strategies:
+ * Credit/Debit Card: 
+ * Cash: 
+
+Interface:
+* Security: The level of protection against fraud or unauthorized transactions.
+* Speed: The efficiency and swiftness of the payment process, including transaction time and waiting in line.
+* Privacy: The confidentiality of personal financial information and transaction details.
```
Comment (ash-teach) [inline README.md]: I don't understand how these bullet points are related to Strategy pattern.

**Review by ash-teach — CHANGES_REQUESTED** [review]

**Comment by ash-teach** [comment]
Thanks for adding this, @jaskaransingh9990. Your example doesn't quite do enough to help explain Strategy: The setup is there (payment choice between Credit, Debit or Cash) but you haven't described how the Strategy pattern helps in this case. Please review and update the PR. Also, there are some typos to be fixed.

**Comment by ash-teach** [comment]
Hm. The changes made don't really address the issue. The overview you've provided is good, but you had some examples that weren't clearly connected with the overview. Removing these and adding code without comment doesn't improve the situation for novices.

**Comment by ash-teach** [comment]
I've removed the code example, since it doesn't really belong in the main README either.

**Pull request merged**